x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/add_source/add_source_logic.test.ts

```diff
@@ -173,70 +222,105 @@ describe('AddSourceLogic', () => {
+        currentServiceType: config.serviceType,
+        githubOrganizations: config.githubOrganizations,
+      });
     });
 
     it('setSelectedGithubOrganizations', () => {
       AddSourceLogic.actions.setSelectedGithubOrganizations('foo');
+      const EXPECTED_BASE_STATE = setupExpectedState();
 
-      expect(AddSourceLogic.values.selectedGithubOrganizationsMap).toEqual({ foo: true });
+      expect(AddSourceLogic.values).toEqual({
+        ...EXPECTED_BASE_STATE,
+        selectedGithubOrganizationsMap: {
+          foo: true,
+        },
+        selectedGithubOrganizations: [ 'foo' ]
+      });
     });
 
     it('setPreContentSourceId', () => {
       AddSourceLogic.actions.setPreContentSourceId('123');
+      const EXPECTED_BASE_STATE = setupExpectedState();
 
-      expect(AddSourceLogic.values.preContentSourceId).toEqual('123');
+      expect(AddSourceLogic.values).toEqual({
+        ...EXPECTED_BASE_STATE,
+        oauthConfigCompleted: false,
+        preContentSourceId: '123',
+      });
     });
 
     it('setButtonNotLoading', () => {
       AddSourceLogic.actions.setButtonNotLoading();
+      const EXPECTED_BASE_STATE = setupExpectedState();
 
-      expect(AddSourceLogic.values.buttonLoading).toEqual(false);
+      expect(AddSourceLogic.values).toEqual({
+        ...EXPECTED_BASE_STATE,
+        buttonLoading: false,
+      });
     });
 
     it('resetSourceState', () => {
       AddSourceLogic.actions.resetSourceState();
 
-      expect(AddSourceLogic.values.dataLoading).toEqual(false);
-      expect(AddSourceLogic.values.buttonLoading).toEqual(false);
-      expect(AddSourceLogic.values.clientIdValue).toEqual('');
-      expect(AddSourceLogic.values.clientSecretValue).toEqual('');
-      expect(AddSourceLogic.values.baseUrlValue).toEqual('');
-      expect(AddSourceLogic.values.loginValue).toEqual('');
-      expect(AddSourceLogic.values.passwordValue).toEqual('');
-      expect(AddSourceLogic.values.subdomainValue).toEqual('');
-      expect(AddSourceLogic.values.indexPermissionsValue).toEqual(false);
-      expect(AddSourceLogic.values.customSourceNameValue).toEqual('');
-      expect(AddSourceLogic.values.newCustomSource).toEqual({});
-      expect(AddSourceLogic.values.currentServiceType).toEqual('');
-      expect(AddSourceLogic.values.githubOrganizations).toEqual([]);
-      expect(AddSourceLogic.values.selectedGithubOrganizationsMap).toEqual({});
+      expect(AddSourceLogic.values).toEqual({
+        ...defaultValues,
+        dataLoading: false,
+        buttonLoading: false,
+        customSourceNameValue: '',
+        clientIdValue: '',
+        clientSecretValue: '',
+        baseUrlValue: '',
+        loginValue: '',
+        passwordValue: '',
+        subdomainValue: '',
+        indexPermissionsValue: false,
+        newCustomSource: {},
+        currentServiceType: '',
+        githubOrganizations: [],
+        selectedGithubOrganizationsMap: {},
+        preContentSourceId: '',
```

Comment:
A lot of these match the default values. We should not assert the ones that are the same. Only those that are different should be here.